### PR TITLE
RPED Patches

### LINKED ISSUE #71801: RPED doesn't exchange datum stock parts correctly & doesn't install datum stock parts
Caused by #71693

Steps to reproduce

1. Create an empty machine frame with anomaly refinery circuit board installed 

2. Create a fully assembled anomaly refinery[by hand cause RPED dosen't work] 

3. Watch the video below

https://user-images.githubusercontent.com/110812394/206229168-8a819c83-0d4d-4316-9570-910061957ba9.mp4

### PR DESCRIPTION
**Pull request by SyncIt21**
## About The Pull Request
Fixes #71801 & also merges changes from  #71775

## Why It's Good For The Game
Bugs. We all hate them & yet we are very good at making them PAIN!!!

## Changelog
:cl:
fix: RPED part display information
fix: RPED incorrectly exchanging parts with the wrong tier for datum stock parts
fix: RPED unable to install datum stock parts in incomplete machine frames
fix: RPED unable to exchange parts for newly constructed machines
/:cl:

### CODE REVIEW

**Comment by SyncIt21** [comment]
@Mothblocks 👍

File: code/modules/research/stock_parts.dm [hunk 1]
```diff
@@ -216,6 +216,15 @@ If you create T5+ please take a pass at mech_fabricator.dm. The parts being good
 	lefthand_file = 'icons/mob/inhands/items/devices_lefthand.dmi'
 	righthand_file = 'icons/mob/inhands/items/devices_righthand.dmi'
 
+/obj/item/storage/part_replacer/proc/get_sorted_parts()
+	var/list/part_list = list()
+	//Assemble a list of current parts, then sort them by their rating!
+	for(var/obj/item/co in contents)
```
Comment (Mothblocks) [inline code/modules/research/stock_parts.dm:222]: Full name--`component`

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/modules/research/stock_parts.dm (hunk 1 above)
Comment (SyncIt21) [reply code/modules/research/stock_parts.dm]: Done

**Review by Mothblocks — APPROVED** [review]

File: code/game/machinery/_machinery.dm [hunk 2]
```diff
@@ -930,6 +930,13 @@
 	if(!machine_board)
 		return FALSE
 
+	/**
+	 * sorting is very important especially because we are breaking the loop on line 1006
```
Comment (Mothblocks) [inline code/game/machinery/_machinery.dm:934]: Please point this to the specific piece of the proc, the lines are going to change.

Comment (SyncIt21) [reply code/game/machinery/_machinery.dm]: Done

**Comment by SyncIt21** [comment]
@Mothblocks RPED was not exchanging parts for newly constructed machines because circuit board was not added to the list of components. Thus exchange_parts() was unable to locate it and did an early return fixed that now.
Also another bug was that if an device required say 2 manipulators and if the RPED only had one the machine the RPED would not remove that installed part from the sorted list after installing causing that one manipulator to replace both in the machine. Thats also fixed now

**Pull request merged**